File: algorithm/challenges/leetcode/1768.merge-strings-alternately/main.clj
(require '[clojure.test :as test])

(defn merge-strings
  [word1 word2]
  (loop [w1 (seq word1)
         w2 (seq word2)
         merged ""]
    (if (or w1 w2)
      (recur w2
             (next w1)
             (str merged (first w1)))
      merged)))

(test/deftest merge-strings-test
  (test/testing "Merge two strings with same size"
    (test/is (= "apbqcr"
                (merge-strings "abc" "pqr"))))
  (test/testing "Merge two strings with different sizes"
    (test/is (= "apbqrs"
                (merge-strings "ab" "pqrs")))))

(test/run-test merge-strings-test)

File: algorithm/challenges/leetcode/1768.merge-strings-alternately/main.py
# %%
def merge_alternately(word1: str, word2: str) -> str:
    """
    Merge two strings
    """
    word1 = list(word1)
    word2 = list(word2)
    merged = []

    while len(word1) + len(word2) > 0:
        if len(word1) > 0:
            merged.append(word1.pop(0))
        if len(word2) > 0:
            merged.append(word2.pop(0))

    return "".join(merged)


merge_alternately("abc", "123")
